analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/Checks/ConstantCheck.cs

```diff
@@ -50,11 +50,12 @@ public static SymbolicConstraint ConstraintFromType(ITypeSymbol type)
             operation.Instance.ConstantValue.Value switch
             {
                 // Update DefaultValue when adding new types
-                true => SymbolicValue.True,
-                false => SymbolicValue.False,
+                true => SymbolicValue.True.WithConstraint(ObjectConstraint.NotNull),
+                false => SymbolicValue.False.WithConstraint(ObjectConstraint.NotNull),
```

Comment:
It might be worth making the combination True/NotNull and False/NotNull a singleton in `SymbolicValue`.
On the other hand, it might be a good idea to change `WithConstraint` instead and cache all the combinations we create to reduce the number of allocations.

Reply:
Yes, this should change the `SymbolicValue.True` and `SymbolicValue.False` fields. So it can be used all over the codebase.